src/ip_notifier.cpp

```diff
@@ -360,40 +359,52 @@ struct ip_change_notifier_impl final : ip_change_notifier
 
 	~ip_change_notifier_impl() override
 	{
-		cancel();
-		// the reason to call close() here is because the
-		// object_handle destructor doesn't close the handle
-		m_hnd.close();
+		if (m_hnd != NULL)
+		{
+			CancelMibChangeNotify2(m_hnd);
+			m_hnd = NULL;
+		}
 	}
 
 	void async_wait(std::function<void(error_code const&)> cb) override
 	{
-		HANDLE hnd;
-		DWORD err = NotifyAddrChange(&hnd, &m_ovl);
-		if (err == ERROR_IO_PENDING)
-		{
-			m_hnd.async_wait([cb](error_code const& ec)
-			{
-				// call CancelIPChangeNotify here?
-				cb(ec);
-			});
-		}
-		else
+		if (m_hnd == NULL)
 		{
-			lt::get_io_service(m_hnd).post([cb, err]()
-			{ cb(error_code(err, system_category())); });
+			cb(make_error_code(boost::system::errc::not_supported));
+			return;
 		}
+
+		std::lock_guard<std::mutex> l(m_cb_mutex);
+		m_cb.emplace_back(std::move(cb));
 	}
 
 	void cancel() override
 	{
-		CancelIPChangeNotify(&m_ovl);
-		m_hnd.cancel();
+		std::vector<std::function<void(error_code const&)>> cbs;
+		{
+			std::lock_guard<std::mutex> l(m_cb_mutex);
+			cbs = std::move(m_cb);
+		}
+		for (auto& cb : cbs) cb(make_error_code(boost::asio::error::operation_aborted));
 	}
 
 private:
-	OVERLAPPED m_ovl = {};
-	boost::asio::windows::object_handle m_hnd;
+	static void address_change_cb(void* ctx, MIB_UNICASTIPADDRESS_ROW*, MIB_NOTIFICATION_TYPE)
+	{
+		ip_change_notifier_impl* impl = static_cast<ip_change_notifier_impl*>(ctx);
+		std::lock_guard<std::mutex> l(impl->m_cb_mutex);
+		impl->m_ios.post([cbs = std::move(impl->m_cb)]()
```

Comment:
`RC_1_2` only requires C++11, so you can't move into lambda (yet) :/

Reply:
(if you leave a comment mentioning C++14, it will be easier to find it once this lands in master)